File: backend/src/main/kotlin/org/globe42/dao/ChargeTypeDao.java
package org.globe42.dao;

import java.util.Optional;

import org.globe42.domain.ChargeType;
import org.springframework.data.jpa.repository.JpaRepository;

/**
 * DAO for the {@link ChargeType} entity
 * @author JB Nizet
 */
public interface ChargeTypeDao extends JpaRepository<ChargeType, Long> {
    boolean existsByName(String name);
    Optional<ChargeType> findByName(String name);
}

File: backend/src/main/kotlin/org/globe42/dao/ChargeTypeDao.kt
package org.globe42.dao

import org.globe42.domain.ChargeType
import org.springframework.data.jpa.repository.JpaRepository

/**
 * DAO for the [ChargeType] entity
 * @author JB Nizet
 */
interface ChargeTypeDao : JpaRepository<ChargeType, Long> {
    fun existsByName(name: String): Boolean
    fun findByName(name: String): ChargeType?
}
